### PR TITLE
Fix pubsub retry for Kafka

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Fix pubsub retry in Kafka.
Add kafka to conf test.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #687

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/conformance/pubsub/pubsub.go [hunk 1]
```diff
@@ -67,6 +70,12 @@ func NewTestConfig(componentName string, allOperations bool, operations []string
 				tc.maxReadDuration = time.Duration(val) * time.Millisecond
 			}
 		}
+		if k == "waitDurationToPublish" {
+			val, err := strconv.Atoi(v)
+			if err == nil {
+				tc.waitDurationToPublish = time.Duration(val) * time.Millisecond
```
Comment (mukundansundar) [inline tests/conformance/pubsub/pubsub.go:76]: initialize with `defaultWaitDurationToPublish` ... thats missing now

File: pubsub/kafka/kafka.go [hunk 2]
```diff
@@ -49,20 +52,39 @@ type kafkaMetadata struct {
 }
 
 type consumer struct {
+	k        *Kafka
```
Comment (mukundansundar) [inline pubsub/kafka/kafka.go:55]: should only `bo` and `logger` be passed on this struct? Why is Kafka struct a pass by ref?

File: tests/config/pubsub/tests.yml [hunk 3]
```diff
@@ -8,4 +8,6 @@ components:
       maxReadDuration: 1000
   - component: redis
     allOperations: true
+  - component: kafka
```
Comment (mukundansundar) [inline tests/config/pubsub/tests.yml:11]: similar config needs to be added to the workflow itself to run it ....

File: pubsub/kafka/kafka.go (hunk 2 above)
Comment (artursouza) [reply pubsub/kafka/kafka.go:55]: Good catch, let me fix that.

File: tests/config/pubsub/tests.yml (hunk 3 above)
Comment (artursouza) [reply tests/config/pubsub/tests.yml:11]: I know. thanks.

File: tests/conformance/pubsub/pubsub.go (hunk 1 above)
Comment (artursouza) [reply tests/conformance/pubsub/pubsub.go:76]: Thanks.

File: tests/conformance/pubsub/pubsub.go [hunk 4]
```diff
@@ -127,7 +152,7 @@ func ConformanceTests(t *testing.T, props map[string]string, ps pubsub.PubSub, c
 	if config.HasOperation("subscribe") {
 		t.Run("verify read", func(t *testing.T) {
 			t.Logf("waiting for %v to complete read", config.maxReadDuration)
-			time.Sleep(config.maxReadDuration)
+			time.Sleep(config.maxReadDuration * 5)
 			assert.LessOrEqual(t, config.messageCount, actualReadCount, "expected to read %v messages", config.messageCount)
```
Comment (mukundansundar) [inline tests/conformance/pubsub/pubsub.go:156]: should this be greater than or equal?

Comment (artursouza) [reply tests/conformance/pubsub/pubsub.go]: This is correct. The issue is in the *5 above that I forgot.

**Comment by artursouza** [comment]
Conf test run is here: https://github.com/dapr/components-contrib/actions/runs/559923872

**Review by yaron2 — APPROVED** [review]

**Comment by mukundansundar** [comment]
e2e test failing fo redelivery in windows https://github.com/dapr/dapr/runs/1884658920?check_suite_focus=true. Trigering again ...

**Pull request merged**